networkit/cpp/centrality/TopCloseness.cpp

```diff
@@ -351,8 +357,10 @@ double TopCloseness::BFScut(node v, double x, std::vector<bool> &visited,
         sum_dist += distances[u];
         if (distances[u] > d) { // Need to update bounds!
             d++;
-            ftildeL = (f + (d + 2) * (rL - nd) - gamma) * (n - 1) / (rL - 1.0) / (rL - 1.0);
-            ftildeU = (f + (d + 2) * (rU - nd) - gamma) * (n - 1) / (rU - 1.0) / (rU - 1.0);
+            ftildeL = (f + static_cast<double>(d + 2) * (rL - nd) - gamma)
+                      * static_cast<double>(n - 1) / (rL - 1.0) / (rL - 1.0);
+            ftildeU = (f + static_cast<double>(d + 2) * (rU - nd) - gamma)
```

Comment:
Shouldn't `(rL - nd)` and `(rU - nd)` also be casted? Also please break the formula.